sway-core/src/asm_generation/fuel/functions.rs

```diff
@@ -66,21 +70,73 @@ impl<'ir, 'eng> FuelAsmBuilder<'ir, 'eng> {
         args: &[Value],
     ) -> Result<(), CompileError> {
         // Put the args into the args registers.
-        for (idx, arg_val) in args.iter().enumerate() {
-            if idx < compiler_constants::NUM_ARG_REGISTERS as usize {
+        if args.len() <= compiler_constants::NUM_ARG_REGISTERS as usize {
```

Comment:
Changing `compiler_constants::NUM_ARG_REGISTERS` will make calling `fn` with different convention an invisible error.

Will this be a problem? Will this affect contract call, library calls etc...

Reply:
What do you mean by a different convention? if someone uses another compiler that uses different conventions for calling, then yes. but otherwise this should be ok. This change impacts all function calls except calls to entry functions (from external sources). Those have a different convention and are already capable of more than NUM_ARG_REGISTERS args.

Reply:
I mean, if the caller uses `NUM_ARG_REGISTERS=2`, for example, and the the callee `NUM_ARG_REGISTERS=6` or whatever, there will be some calls that will error.

Caller is putting the third argument in the stack, but the callee is actually expecting it as a register.

Just wondering how hard/impactful would be to change these things.

Reply:
>if the caller uses `NUM_ARG_REGISTERS=2`, for example, and the the callee `NUM_ARG_REGISTERS=6`

That would only happen if the caller and the callee were separately compiled with different (versions of the) compiler(s). I'm guessing that that's not a scenario we need to worry about.

Reply:
This wouldn't be able to happen except through contract calls which use a different convention, correct @vaivaswatha ?

Reply:
That's right, yes. 

We don't have a "linking" phase. We compile the whole program every time. So the whole program will use the same calling convention. And as you mentioned, entry points have a different convention altogether.

Reply:
That was my concern. If this is not our use case then I am fine. :)